ludwig/models/predictor.py

```diff
@@ -193,15 +194,16 @@ def batch_evaluation(self, dataset, collect_predictions=False, collect_logits=Fa
                         f"evaluation for {dataset_name}: obtained next batch "
                         f"memory used: {psutil.Process(os.getpid()).memory_info()[0] / 1e6:0.2f}MB"
                     )
+                    # TODO (jeffkinnison): revert to using the requested device for GBMs when device usage is fixed
                     inputs = {
                         i_feat.feature_name: torch.from_numpy(np.array(batch[i_feat.proc_column], copy=True)).to(
-                            self.device
+                            self.device if not isinstance(self.model, GBM) else "cpu"
```

Comment:
Would rather have this be a param we could pass in to the batch evaluator, but okay with something like this if it's only meant to be very temporary.

Can you at least pull this out to the init of `self.device` instead, though?

Reply:
Sure, I can do that. This is only meant to be a temp fix so GBM eval doesn't fail when a GPU is present. I definitely want to clean this up more generally.